src/main/java/entity_layer/CommonTaskTree.java

```diff
@@ -54,7 +107,19 @@ public boolean removeChildTaskTree(TaskTree taskTree) {
      */
     @Override
     public TaskTree getChildTaskTreeByTask(Task task) {
-        return null;
+        // Check if this is the matching tree
+        if (this.task == task)
```

Comment:
This implementation might be problematic because '==' works differently on objects in Java. This will only return true if the given task and the compared have the same address in memory. You might need to use .equals() and overwrite it for tasks.
https://www.baeldung.com/java-comparing-objects

Reply:
That's more or less how I wanted it to be implemented, in fact. To compare the actual Task objects in memory. However, looking at it now, I may forego these methods entirely, as they really aren't going to be useful for the interactors since they are only given the taskID's.